### ISSUE TITLE
Huge actor loss in `TD3`

### ISSUE THREAD

**Issue by AndreasKaratzas** [DLR-RM/stable-baselines3#937]
### Question

I am using  a `TD3` agent to solve a `MountainCarContinuous-v0` environment. Is it normal that the actor's loss gets huge at some point? Isn't this dangerous for the weights of the model? The weird part is that the agent is actually solving the environment, since the accumulated reward is gradually increasing.

### Additional context

This is my code (as suggested by @qgallouedec in https://github.com/DLR-RM/stable-baselines3/issues/936):

```python
import gym
import numpy as np
from stable_baselines3 import TD3
from stable_baselines3.common.noise import OrnsteinUhlenbeckActionNoise

env = gym.make("MountainCarContinuous-v0")

action_noise = OrnsteinUhlenbeckActionNoise(np.zeros(1), 0.5 * np.ones(1))
model = TD3("MlpPolicy", env, action_noise=action_noise, verbose=1)

model.learn(total_timesteps=1_000_000)
```

This is the logger's output after some iterations.

```
---------------------------------
| rollout/           |          |
|    ep_len_mean     | 361      |
|    ep_rew_mean     | 60.8     |
| time/              |          |
|    episodes        | 68       |
|    fps             | 111      |
|    time_elapsed    | 219      |
|    total_timesteps | 24547    |
| train/             |          |
|    actor_loss      | -14.4    |
|    critic_loss     | 1.08     |
|    learning_rate   | 0.001    |
|    n_updates       | 24471    |
---------------------------------
```

**Comment by qgallouedec** [comment]
What is "huge" and "at some point" for you?

**Comment by AndreasKaratzas** [comment]
"Huge" is a loss that exceeds 10.0, like in the attached logger's output.
"At some point": really depends on the seed and on other training parameters (like the noise factor) that result in the agent's learning the environment. For example, in the attached logger's output, that point is close to the 70th episode.

My definition of large actor loss might be "something greater than 10" by the way, but it is amplified as the agent is training. For example, it gets close to 20 in episode 80 and sometimes it even exceeds 50.

**Comment by araffin** [comment]
Hello,
you should probably read more about DDPG/TD3 (for instance https://spinningup.openai.com/en/latest/algorithms/ddpg.html), the actor loss function is not a "classic" loss, so its magnitude (except if its infinity) and monotony does not tell you much about if the agent is learning something or not.

The critic loss is a bit more informative but what really matter is the episodic return (the actual performance of the algorithm in the env).

**Comment by ldesmet** [comment]
Hello,
In different environment I got acot loss which is way bigger (around 100 even 1000 sometimes) but the agent is still learning. 
I would say it depends on cumulative reward in your environment...

**Comment by araffin** [comment]
>  it depends on cumulative reward in your environment...

yes, that's how the actor loss is defined:
https://github.com/DLR-RM/stable-baselines3/blob/a6f5049a99a4c21a6f0bcce458ca3306cef310e0/stable_baselines3/td3/td3.py#L182

It depends on the magnitude of the reward and on the discount factor.

**Comment by AndreasKaratzas** [comment]
I understand that the true agent progress metric is the episode return, but how does such a huge loss not damage the model? Should the environment's states be somehow normalized or rescaled I such cases? I mean if a big loss is continuously propagated to a model, then the model will have very large weights resulting sometimes in numerical instability, right?

**Comment by araffin** [comment]
> Should the environment's states be somehow normalized or rescaled I such cases?

https://stable-baselines3.readthedocs.io/en/master/guide/rl_tips.html#tips-and-tricks-when-creating-a-custom-environment

> I mean if a big loss is continuously propagated to a model, then the model will have very large weights resulting sometimes in numerical instability, right?

Not necessarily. You are taking the gradient of the loss and the update magnitude also depends on the learning rate.

As a quick example, let say your loss is $x^2 + 10000$, your loss magnitude may be high but the gradient w.r.t. x is only $2 *x$.

**Comment by AndreasKaratzas** [comment]
Yeah, correct. Well, I think that covers my question. Thank you all :)
